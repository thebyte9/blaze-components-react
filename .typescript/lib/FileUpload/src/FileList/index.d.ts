/// <reference types="react" />
declare const FileList: ({ previewImages, handleCancel, handleInputChange, handleSelectChange, selectOptions }: {
    previewImages: any;
    handleCancel: any;
    handleInputChange: any;
    handleSelectChange: any;
    selectOptions: any[];
}) => JSX.Element;
export default FileList;
