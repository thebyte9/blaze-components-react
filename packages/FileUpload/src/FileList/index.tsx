import React, { Fragment } from 'react';
import { DOC, IMAGE } from '../constants';
import DocumentIcon from '../DocumentIcon';
import FileInputs from '../FileInputs';

const FileList = ({
  previewImages,
  handleCancel,
  handleInputChange,
  handleSelectChange,
  selectOptions,
}: {
  previewImages: any;
  handleCancel: any;
  handleInputChange: any;
  handleSelectChange: any;
  selectOptions: any[];
}) => (
  <>
    {previewImages.map(
      (
        {
          file,
          name,
          data,
        }: {
          file: any;
          name: string;
          data: any | null;
        },
        index: any,
      ) => {
        const fileInputsProps = {
          data,
          file,
          handleInputChange,
          handleSelectChange,
          index,
          name,
          selectOptions,
        };
        const isImage = file.type === IMAGE;

        return (
          <Fragment key={file.id}>
            <div className="preview">
              <div className={`preview__file preview__file--${file.type}`}>
                {isImage ? <img src={file.base64} /> : <DocumentIcon />}
              </div>
              <div className="preview__details">
                <div className="preview__filename">{file.name}</div>
                <div className="preview__form">
                  {(isImage || file.type === DOC) && <FileInputs {...fileInputsProps} />}
                  <i
                    onClick={() => handleCancel(file.id)}
                    className="material-icons"
                    aria-hidden="true"
                    data-testid="delete"
                  >
                    delete_outline
                  </i>
                </div>
              </div>
            </div>
          </Fragment>
        );
      },
    )}
  </>
);

export default FileList;
