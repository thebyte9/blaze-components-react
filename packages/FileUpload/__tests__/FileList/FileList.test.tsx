import FileList from '../../src/FileList';
import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { nanoid } from 'nanoid';

const handleCancelMock = jest.fn();
const handleInputChangeMock = jest.fn();
const handleSelectChange = jest.fn();

const FileListComponent = ({ previewImages }) => (
  <FileList
    previewImages={previewImages}
    handleCancel={handleCancelMock}
    handleInputChange={handleInputChangeMock}
    handleSelectChange={handleSelectChange}
    selectOptions={[]}
  />
);

describe('FileList component', () => {
  test('should match snapshot', () => {
    const previewImages = [
      {
        file: {
          id: nanoid(),
          base64: new Blob(['image content'], { type: 'image/png' }),
          type: 'image',
          name: 'file-1.png',
        },
        data: { caption: 'test caption' },
      },
    ];
    const { asFragment } = render(FileListComponent({ previewImages }));
    expect(asFragment()).toMatchSnapshot();
  });

  test('should fire delete event', () => {
    const previewImages = [
      {
        file: {
          id: nanoid(),
          base64: new Blob(['doc content'], { type: 'doc' }),
          type: 'doc',
          name: 'file-2.doc',
        },
        data: { caption: 'test caption' },
      },
    ];
    render(FileListComponent({ previewImages }));
    fireEvent.click(screen.getByTestId('delete'));
  });

  test('should fire onChange event', () => {
    const previewImages = [
      {
        file: {
          id: nanoid(),
          base64: new Blob(['image content'], { type: 'image/png' }),
          type: 'image',
          name: 'file-1.png',
        },
        data: { caption: 'test caption' },
      },
    ];
    render(FileListComponent({ previewImages }));
    fireEvent.change(screen.getByTestId('store-type'));
  });

  test('should return input label for video files ', () => {
    const previewImages = [
      {
        file: {
          id: nanoid(),
          base64: new Blob(['image content'], { type: 'video/mp4' }),
          type: 'video',
          name: 'video.mp4',
        },
        data: { caption: 'test caption' },
      },
    ];
    const { asFragment } = render(FileListComponent({ previewImages }));
    expect(asFragment()).toMatchSnapshot();
  });
});
