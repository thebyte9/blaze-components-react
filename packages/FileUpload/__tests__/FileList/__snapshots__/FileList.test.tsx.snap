// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileList component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="preview"
  >
    <div
      class="preview__file preview__file--image"
    >
      <img
        src="[object Blob]"
      />
    </div>
    <div
      class="preview__details"
    >
      <div
        class="preview__filename"
      >
        file-1.png
      </div>
      <div
        class="preview__form"
      >
        <div
          class="form-field form-field--input  "
        >
          <label
            class=""
            for="0-title-file-1png"
          >
            image title
          </label>
          <input
            data-testid="input"
            id="0-title-file-1png"
            name="name"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-field form-field--input  "
        >
          <label
            class=""
            for="0-caption-file-1png"
          >
            image caption
          </label>
          <input
            data-testid="input"
            id="0-caption-file-1png"
            name="caption-0-file-1png"
            type="text"
            value="test caption"
          />
        </div>
        <div
          class="form-field form-field--input  "
        >
          <label
            class=""
            for="0-altText-file-1png"
          >
            Alternative text
          </label>
          <input
            data-testid="input"
            id="0-altText-file-1png"
            name="altText-0-file-1png"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-field form-field--input  "
        >
          <label
            class=""
            for="0-hrefUrl-file-1png"
          >
            Url
          </label>
          <input
            data-testid="input"
            id="0-hrefUrl-file-1png"
            name="hrefUrl-0-file-1png"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-field form-field--select"
        >
          <label
            class=""
          >
            Store type
          </label>
          <select
            data-testid="store-type"
            disabled=""
          >
            <option>
              Please Choose...
            </option>
          </select>
        </div>
        <i
          aria-hidden="true"
          class="material-icons"
          data-testid="delete"
        >
          delete_outline
        </i>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FileList component should return input label for video files  1`] = `
<DocumentFragment>
  <div
    class="preview"
  >
    <div
      class="preview__file preview__file--video"
    >
      <svg
        height="100"
        viewBox="0 0 8 8"
        width="100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0v8h7v-4h-4v-4h-3zm4 0v3h3l-3-3zm-3 2h1v1h-1v-1zm0 2h1v1h-1v-1zm0 2h4v1h-4v-1z"
        />
      </svg>
    </div>
    <div
      class="preview__details"
    >
      <div
        class="preview__filename"
      >
        video.mp4
      </div>
      <div
        class="preview__form"
      >
        <i
          aria-hidden="true"
          class="material-icons"
          data-testid="delete"
        >
          delete_outline
        </i>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
